internal/resources/grafana/resource_alerting_rule_group_test.go

```diff
@@ -675,6 +675,34 @@ func TestAccAlertRule_NotificationSettings(t *testing.T) {
 	})
 }
 
+func TestAccRecordingRule(t *testing.T) {
+	testutils.CheckOSSTestsEnabled(t, ">=11.2.0")
+
+	var group models.AlertRuleGroup
+	var name = acctest.RandString(10)
+	var metric = acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProtoV5ProviderFactories: testutils.ProtoV5ProviderFactories,
+		CheckDestroy:             alertingRuleGroupCheckExists.destroyed(&group, nil),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRecordingRule(name, metric, "A"),
+				Check: resource.ComposeTestCheckFunc(
+					alertingRuleGroupCheckExists.exists("grafana_rule_group.my_rule_group", &group),
+					resource.TestCheckResourceAttr("grafana_rule_group.my_rule_group", "name", name),
+					resource.TestCheckResourceAttr("grafana_rule_group.my_rule_group", "rule.#", "1"),
+					resource.TestCheckResourceAttr("grafana_rule_group.my_rule_group", "rule.0.name", "My Random Walk Alert"),
+					resource.TestCheckResourceAttr("grafana_rule_group.my_rule_group", "rule.0.for", "2m0s"),
+					resource.TestCheckResourceAttr("grafana_rule_group.my_rule_group", "rule.0.data.0.model", "{\"hide\":false,\"refId\":\"A\"}"),
+					resource.TestCheckResourceAttr("grafana_rule_group.my_rule_group", "rule.0.data.0.record.metric", metric),
+					resource.TestCheckResourceAttr("grafana_rule_group.my_rule_group", "rule.0.data.0.record.from", "A"),
+				),
```

Comment:
Can we also get corresponding tests for Err/NoData state and the other alert rule fields? Even if it just asserts it's properly ignored.

Maybe the provider should reject such rules, that define these fields on recording rules?